File: quizzes/hackerrank/fp/introduction/hello_world_n_times.hs
#!/usr/bin/runhaskell
-- author: jinchoiseoul@gmail.com

import Control.Applicative
import Control.Monad
import System.IO

main :: IO ()
main = do
    n_temp <- getLine
    let n = read n_temp :: Int
    --  Print "Hello World" on a new line 'n' times.
    mapM_ putStrLn (getMultipleLines n)
    
getMultipleLines :: Int -> [String]

getMultipleLines n
    | n <= 0 = []
    | otherwise = "Hello World" : getMultipleLines (n-1)

File: quizzes/hackerrank/fp/introduction/hello_world_n_times.scala
#!/usr/bin/env scala
// author: jinchoiseoul@gmail.com

object Solution {
    def f(n : Int) {
        for (i <- 1 to n)
            println("Hello World");
    }

    def main(args: Array[String]) {
        val n = scala.io.StdIn.readInt;
        f(n);
    }
}
